### PR TITLE
MAT-8909 library lock

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-8909](https://jira.cms.gov/browse/MAT-8909)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included and Code coverage is verified.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/cqllibraryservice/locks/CqlLibraryLock.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Document(collection = "cqlLibraryLock")
+public class CqlLibraryLock {
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/cqllibraryservice/locks/CqlLibraryLock.java:19]: Is there a reason why we are not using madie-java-models ?

File: src/test/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryLockControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  @Test
+  public void testAddCqlLibraryLock() {
+    Principal principal = mock(Principal.class);
+    when(principal.getName()).thenReturn("test.user");
+    when(service.lockCqlLibrary(anyString(), anyString())).thenReturn(lockInfo);
+
+    ResponseEntity<LockInfo> response = controller.addCqlLibraryLock("cqlLibrayId", principal);
+    assertNotNull(response);
+    assertEquals(response.getBody().getLockedBy(), "test.user");
```
Comment (RohitKandimalla) [inline src/test/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryLockControllerTest.java]: Looks like Arguments to 'assertEquals()' are in wrong order through all the test cases.
It should be     assertEquals("test.user", response.getBody().getLockedBy());

File: src/main/java/gov/cms/madie/cqllibraryservice/locks/CqlLibraryLock.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/cqllibraryservice/locks/CqlLibraryLock.java:19]: @RohitKandimalla I think it started with the measure lock implementation: https://github.com/MeasureAuthoringTool/madie-java-models/pull/300, https://github.com/MeasureAuthoringTool/measure-service/pull/921, so later on I followed the same pattern.

File: src/test/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryLockControllerTest.java (hunk 2 above)
Comment (sb-cecilialiu) [reply src/test/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryLockControllerTest.java]: @RohitKandimalla thanks Rohit i changed the order.

**Review by RohitKandimalla — APPROVED** [review]
